### ISSUE TITLE
2.4.11: Gradient Backgrounds

### ISSUE THREAD

**Issue by benja11y** [w3c/wcag#1862]
The [understanding document](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible-enhanced) goes into detail on how to approach gradational focus indicators, but no examples have been given describing how to handle gradational backgrounds.  Pages that have a non-solid background image covering the whole page or make use of parallax scrolling effects will result in a near-infinite number of color combinations if a page is scrolled and/or changes are made to the viewport size.

How can this be addressed pragmatically to benefit users without causing undue burden on testers at the same time?

This issue has been raised on behalf of [TPGi](https://www.tpgi.com), based on an internal consultation process. Links to all of our issues will shortly be found on TPGi's inclusion blog along with an introduction and acknowledgments.  This paragraph will be replaced when the blog post has been published.

**Comment by alastc** [comment]
Would it not be the same as for text-contrast for text on a gradient background?

I.e. there isn't a programmatic method (yet), but it is fairly easy to manipulate the page to find the worst-case, and test that.

**Comment by mraccess77** [comment]
My personal opinion is that if the contrast of background colors that change are close enough to need to be tested for each combination then they would likely not meet the user need of people with low vision in certain scroll combinations and would likely fail in certain combinations as well.  I agree that the text would have the same issue today under SC 1.4.3.  In these cases it would be an easy solution to use a double ring focus indicator or some other mechanism to indicate focus such as a solid box with a border to guarantee there is sufficient contrast across variations of background images or background gradients.

**Comment by DavidMacDonald** [comment]
Agree with @mraccess77 that this is the same issue as 1.4.3 currently.

I suggest adding the following to the Understanding:

> If pages that have a non-solid background image covering the whole page or make use of parallax scrolling effects which result in a near-infinite number of color combinations if a page is scrolled and/or changes are made to the viewport size and if the contrast of background colors that change are close enough to need to be tested for each combination then they would likely not meet the user need of people with low vision in certain scroll combinations and would likely fail in certain combinations as well. In these cases it would be an easy solution to use a double ring focus indicator or some other mechanism to indicate focus such as a solid box with a border to guarantee there is sufficient contrast across variations of background images or background gradients.
